js/classes/game_engine.js

```diff
@@ -74,31 +94,46 @@ class GameEngine {
      * This method is going to go through all entities and allow them to update their position.
      */
     update() {
-        // Update the HUD and Crosshair regardless of whether the game is running or not
-        HUD.update();
-        CROSSHAIR.update();
-
         if (this.running) {
-            // Iterate forward through the entities array. Update everything.
-            for (let i = 0; i < this.entities.length; i++) {
-                let entity = this.entities[i];
-                if (!entity.removeFromWorld) {
+            // (1) Update the background entities:
+            this.entities.background.forEach((entity) => {
+                if (entity.removeFromWorld) {
+                    const i = this.entities.background.indexOf(entity);
+                    this.entities.background.splice(i, 1);
+                } else {
                     entity.update();
                 }
-            }
-            // Update Chad, who is not a regular entity.
+            });
+
+            // (2) Update the midground entities:
+            this.entities.midground.forEach((entity) => {
+                if (entity.removeFromWorld) {
+                    const i = this.entities.midground.indexOf(entity);
+                    this.entities.midground.splice(i, 1);
+                } else {
+                    entity.update();
+                }
+            });
+
+            // (3) Update CHAD and CAMERA:
             CHAD.update();
-            // Update the camera, which is not a regular entity.
             CAMERA.update();
 
-            // Iterate backward through the entities array, and remove all entities which ought be removed.
-            for (let i = this.entities.length - 1; i >= 0; i--) {
-                if (this.entities[i].removeFromWorld) {
-                    this.entities.splice(i, 1);
+            // (4) Update the foreground entities:
+            this.entities.foreground.forEach((entity) => {
```

Comment:
We could probably generalize these forEach loops into a function that accepts the fore/mid/background array as a parameter to simplify this code.

Reply:
Seems like a popular opinion. I'll look into it.